src/canard.h

```diff
@@ -5,9 +29,187 @@
 extern "C" {
 #endif
 
+#include <stdio.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+#include <stdbool.h>
+
 /** The size of a memory block in bytes. */
 #define CANARD_MEM_BLOCK_SIZE 32
 
+#define CANARD_CAN_FRAME_MAX_DATA_LEN 8
+
+#define CANARD_BROADCAST_NODE_ID    0
+#define CANARD_MIN_NODE_ID          1
+#define CANARD_MAX_NODE_ID          127
+
+#define CANARD_RX_PAYLOAD_HEAD_SIZE (CANARD_MEM_BLOCK_SIZE - sizeof(CanardRxState))
+#define CANARD_BUFFER_BLOCK_DATA_SIZE (CANARD_MEM_BLOCK_SIZE - sizeof(CanardBufferBlock))
+
+typedef struct
+{
+  uint32_t id;
+  uint8_t data[CANARD_CAN_FRAME_MAX_DATA_LEN];
+  uint8_t data_len;
+} CanardCANFrame;
+
+typedef enum
+{
+  CanardTransferTypeResponse  = 0,
+  CanardTransferTypeRequest   = 1,
+  CanardTransferTypeBroadcast = 2
+} CanardTransferType;
+
+typedef enum
+{
+  CanardResponse,
+  CanardRequest
+} CanardRequestResponse;
+
+typedef struct CanardInstance CanardInstance;
+typedef struct CanardRxTransfer CanardRxTransfer;
+typedef struct CanardRxState CanardRxState;
+typedef struct CanardTxQueueItem CanardTxQueueItem;
+
+/**
+ * The library calls this function to determine whether the transfer should be received.
+ */
+typedef bool (*canardShouldAcceptTransferPtr)(const CanardInstance* ins,
+                                                uint64_t* out_data_type_signature,
+                                                uint16_t data_type_id, 
+                                                CanardTransferType transfer_type, 
+                                                uint8_t source_node_id);
+
+/**
+ * This function will be invoked by the library every time a transfer is successfully received.
+ * If the application needs to send another transfer from this callback, it is recommended
+ * to call canardReleaseRxTransferPayload() first, so that the memory that was used for the block
+ * buffer can be released and re-used by the TX queue.
+ */
+typedef void (*canardOnTransferReception)(CanardInstance* ins, CanardRxTransfer* transfer);
```

Comment:
Type names are supposed to be named in Pascal Case (in the design document it's named incorrectly): `CanardOnTransferReception`